.github/workflows/ci.yml

```diff
@@ -359,42 +359,39 @@ jobs:
             **/surefire-reports/TEST-*.xml
           retention-days: ${{ env.TEST_REPORT_RETENTION_DAYS }}
 
+  build-test-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base for GIB
+      - name: Fetch base ref
+        run: |
+          [ "$GITHUB_REF_NAME" != master ] && git fetch --no-tags --prune --progress --no-recurse-submodules origin "$GITHUB_BASE_REF"
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: 11
+          cache: 'maven'
+      - name: Maven install
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $RETRY $MAVEN install ${MAVEN_FAST_INSTALL} -pl '!:trino-docs,!:trino-server,!:trino-server-rpm'
+      - id: set-matrix
+        run: |
+          ./.github/bin/build-matrix-from-impacted.py -v -o matrix.json
+          echo "Matrix: $(jq '.' matrix.json)"
+          echo "::set-output name=matrix::$(cat matrix.json)"
+
   test:
     runs-on: ubuntu-latest
+    needs: build-test-matrix
+    if: needs.build-test-matrix.outputs.matrix != '{}'
     strategy:
       fail-fast: false
-      matrix:
-        include:
```

Comment:
We should keep the list inside this file and pass it as an argument to the matrix action, because it's important that the exclusion list in the ` test-other-modules` step matches, since the goal of that is to test all the modules that aren't tested separately, and because having it external makes it harder to find. Ideally, we could generate the exclusion list automatically based on the module list.